### PR TITLE
Updates to Windows CI

### PR DESCRIPTION
**Pull request by cmb69**
We
* switch to ~~the latest commit~~ <ins>v0.10</ins> of php/setup-php-sdk
* we enable caching
* we switch to downloads.php.net
* we work around the missing libmemcached-vs17
* we enable testing for PHP 8.4 on Windows

### CODE REVIEW

File: .github/workflows/build-windows.yml [hunk 1]
```diff
@@ -7,23 +7,30 @@ jobs:
...
         with:
           version: ${{matrix.version}}
           arch: ${{matrix.arch}}
           ts: ${{matrix.ts}}
           deps: zlib
+          cache: true
       - name: Fetch libmemcached
-        run: curl -OLs https://windows.php.net/downloads/pecl/deps/libmemcached-1.1.1-${{steps.setup-php.outputs.vs}}-${{matrix.arch}}.zip && 7z x libmemcached-1.1.1-${{steps.setup-php.outputs.vs}}-${{matrix.arch}}.zip -o..\deps
+        run: |
+          set MEMCACHED_FILENAME=libmemcached-1.1.1-${{steps.setup-php.outputs.vs == 'vs17' && 'vs16' || steps.setup-php.outputs.vs}}-${{matrix.arch}}.zip
```
Comment (m6w6) [inline .github/workflows/build-windows.yml:32]: Why 1.1.1 from 2021?
libmemcached is [currently at 1.1.4](https://github.com/awesomized/libmemcached/releases/tag/1.1.4) which actually contains a fix for a [possible security issue](https://github.com/awesomized/libmemcached/security/advisories/GHSA-wwmh-39wj-fx59).
I totally understand, though, if the answer is obviously manpower, etc...

**Comment by m6w6** [comment]
Also, the 3.3.0RC1 DLL links on PECL seem broken, is this a fix for this, too?

**Comment by cmb69** [comment]
> Also, the 3.3.0RC1 DLL links on PECL seem broken, […]

The URL is https://downloads.php.net/~windows/pecl/releases/memcached/3.3.0RC1/php_memcached-3.3.0rc1-8.3-ts-vs16-x64.zip, but should be https://downloads.php.net/~windows/pecl/releases/memcached/3.3.0RC1/php_memcached-3.3.0RC1-8.3-ts-vs16-x64.zip (note lower-case vs. upper-case "RC").

cc @derickr, @shivammathur

File: .github/workflows/build-windows.yml (hunk 1 above)
Comment (cmb69) [reply .github/workflows/build-windows.yml:32]: > Why 1.1.1 from 2021?

Must be one of the more recent dependency builds. Anyhow, let's see how https://github.com/winlibs/winlib-builder/actions/runs/11332741118 goes.

Comment (cmb69) [reply .github/workflows/build-windows.yml:32]: Looks like I'll have to revert https://github.com/awesomized/libmemcached/commit/1a15a0895b863a02d598a317f4c0215edef12dc5 to get back to PHP Windows naming conventions.

**Comment by shivammathur** [comment]
@cmb69 @m6w6  The PECL DLL links should be fixed now.

File: .github/workflows/build-windows.yml (hunk 1 above)
Comment (m6w6) [reply .github/workflows/build-windows.yml:32]: That commit looks like a fix to not end up with liblibfoo.a
What's the exact problem on your side?
I guess we could fix both sides if we get rid of the ALIAS shenanigans.

Comment (cmb69) [reply .github/workflows/build-windows.yml:32]: Without the patch, on Windows memcached.lib is built; with the fix it's libmemcached.lib. The former is probably common on Windows, but the latter is the PHP convention for import libraries (would be libmemcached_a.lib for static libs). If you "fix" this, others building the library might not get the library names they want/expect.

Anyway, applying a patch is not hard, and [the build was successful](https://github.com/winlibs/winlib-builder/actions/runs/11343821419). I've built ext/memcached locally successfully against libmemcached 1.1.4.

The actual problems are lack of automation of doing and testing dependency builds, and of course, that we can hardly watch for new dependency releases (way too many packages). And some packages need way more heavy patching anyway.

Now, someone needs to upload the [libmemcached-1.1.4 builds](https://github.com/winlibs/winlib-builder/actions/runs/11343821419) to the download server. Maybe @shivammathur is in the mood. :)

Comment (cmb69) [reply .github/workflows/build-windows.yml:32]: PR to use it for Windows CI: #564.

Comment (m6w6) [reply .github/workflows/build-windows.yml:32]: 🥳

**Pull request merged**